src/new-client/src/App/Trades/TradesGrid/TradesGrid.tsx

```diff
@@ -145,48 +146,70 @@ const DATE_FORMAT = "dd-MMM-yyyy"
 
 const formatTo6Digits = significantDigitsNumberFormatter(6)
 
-export const TradesGrid: React.FC = () => {
-  const trades = useTableTrades()
+interface RowParam {
+  index: number
+  style: unknown
+}
+
+const MainTable: React.FC = ({ children }) => {
   const tableSort = useTableSort()
   return (
-    <TableWrapper>
-      <Table>
-        <TableHead>
-          <TableHeadRow>
-            <StatusIndicatorSpacer />
-            {colConfigs.map(({ field, headerName, numeric }) => (
-              <TableHeadCell
-                key={field}
-                numeric={numeric}
-                onClick={() => sortFieldSelections$.next(field)}
-              >
-                {headerName}
-                {tableSort.field === field ? (
-                  tableSort.direction === "ASC" ? (
-                    <FaLongArrowAltUp />
-                  ) : (
-                    <FaLongArrowAltDown />
-                  )
+    <Table>
+      <TableHead>
+        <TableHeadRow>
+          <StatusIndicatorSpacer />
+          {colConfigs.map(({ field, headerName, numeric }) => (
+            <TableHeadCell
+              key={field}
+              numeric={numeric}
+              onClick={() => sortFieldSelections$.next(field)}
+            >
+              {headerName}
+              {tableSort.field === field ? (
+                tableSort.direction === "ASC" ? (
+                  <FaLongArrowAltUp />
                 ) : (
-                  <span className="spacer" />
-                )}
-              </TableHeadCell>
-            ))}
-          </TableHeadRow>
-        </TableHead>
-        <tbody>
-          {trades.map((trade) => (
-            <TableBodyRow key={trade.tradeId}>
-              <StatusIndicator status={trade.status} />
-              {colConfigs.map(({ field, numeric, valueFormatter }) => (
-                <TableBodyCell key={field} numeric={numeric}>
-                  {valueFormatter?.(trade[field]) ?? trade[field]}
-                </TableBodyCell>
-              ))}
-            </TableBodyRow>
+                  <FaLongArrowAltDown />
+                )
+              ) : (
+                <span className="spacer" />
+              )}
+            </TableHeadCell>
           ))}
-        </tbody>
-      </Table>
+        </TableHeadRow>
+      </TableHead>
+      {children}
+    </Table>
+  )
+}
+
+export const TradesGrid: React.FC = () => {
+  const trades = useTableTrades()
+
+  const Row = ({ index, style }: RowParam) => (
```

Comment:
Since this doesn't depend directly on any state from the TradesGrid, you can/should declare it outside of the TradesGrid component.  That way the Row component is created once, rather than on every render.  Also, following the naming pattern, call it "VirtualizedTableRow"

Reply:
I tried to do it yesterday but how can I help the Row component to consume the Rxjs observable outside the components? Should I call the useTableTrades in Row component?

Reply:
From what I read in the docs just how, I believe you pass the trades array into the List/VirtualizedTable via the `itemData` prop.  It's then available to your row via the 'data' prop.  So you should be able to access each trade inside the row component as  `data[index]`.  Because it's typed as an `any` you'll like have to cast it as `Trade`.